pkg/authenticator/target_cluster_client.go

```diff
@@ -76,3 +82,34 @@ func (kc *targetClusterClient) GetKubeconfigString(ctx context.Context, clusterN
 
 	return kubeconfigSecret.Data["value"], nil
 }
+
+func (kc *targetClusterClient) GetServerVersion(ctx context.Context, clusterName string) (info *version.Info, err error) {
```

Comment:
Let me think about this. I'm not a big fan of the idea of creating these statically because that would mean we'd have to manage those resources rather than just creating them on the fly. I do like the idea of rearranging this code a bit though. Part of the reason for the awkwardness here is the helm implementation requiring a file. I was working on a work-around for that, but never finished it. I think if I brought that in again, that might clean up those concerns.